crates/anvil/core/src/eth/utils.rs

```diff
@@ -1,43 +1,44 @@
-use alloy_eips::eip2930::AccessListItem as AlloyEipAccessListItem;
-use alloy_primitives::{Address, U256};
-use alloy_rpc_types::AccessListItem as AlloyAccessListItem;
-use ethers_core::{
-    types::transaction::eip2930::AccessListItem,
-    utils::{
-        rlp,
-        rlp::{Encodable, RlpStream},
-    },
+use alloy_eips::eip2930::{
+    AccessList as AlloyEipAccessList, AccessListItem as AlloyEipAccessListItem,
 };
-use foundry_common::types::ToAlloy;
-
-pub fn enveloped<T: Encodable>(id: u8, v: &T, s: &mut RlpStream) {
-    let encoded = rlp::encode(v);
-    let mut out = vec![0; 1 + encoded.len()];
-    out[0] = id;
-    out[1..].copy_from_slice(&encoded);
-    out.rlp_append(s)
-}
-
-pub fn to_revm_access_list(list: Vec<AccessListItem>) -> Vec<(Address, Vec<U256>)> {
-    list.into_iter()
-        .map(|item| {
-            (
-                item.address.to_alloy(),
-                item.storage_keys.into_iter().map(|k| k.to_alloy().into()).collect(),
-            )
-        })
-        .collect()
-}
+use alloy_primitives::{Address, Parity, U256};
+use alloy_rpc_types::{AccessList as AlloyAccessList, AccessListItem as AlloyAccessListItem};
 
 pub fn alloy_to_revm_access_list(list: Vec<AlloyAccessListItem>) -> Vec<(Address, Vec<U256>)> {
     list.into_iter()
         .map(|item| (item.address, item.storage_keys.into_iter().map(|k| k.into()).collect()))
         .collect()
 }
 
+/// Translates a vec of [AlloyEipAccessListItem] to a [AlloyAccessList], translating from internal
+/// type to rpc type.
+pub fn from_eip_to_alloy_access_list(list: AlloyEipAccessList) -> AlloyAccessList {
+    AlloyAccessList(
+        list.0
+            .into_iter()
+            .map(|item| AlloyAccessListItem {
+                address: item.address,
+                storage_keys: item.storage_keys.into_iter().collect(),
+            })
+            .collect(),
+    )
+}
+
 /// Translates a vec of [AlloyEipAccessListItem] to a revm style Access List.
 pub fn eip_to_revm_access_list(list: Vec<AlloyEipAccessListItem>) -> Vec<(Address, Vec<U256>)> {
     list.into_iter()
         .map(|item| (item.address, item.storage_keys.into_iter().map(|k| k.into()).collect()))
         .collect()
 }
```

Comment:
this conversion shouldn't be necessary, let's try to use compatible types?

Reply:
Right now they are—consensus txs use the `alloy_eips` access list type, not the rpc_types one. I do think there's an opportunity to unify both crates somehow as we talked in the sync earlier in the week. I'll file an issue for this

Reply:
https://github.com/alloy-rs/alloy/issues/147